### PR TITLE
Add rejection sampling script

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR adds the rejection sampling script. For demo, see 

https://drive.google.com/file/d/1vxhgtWbjXFOWBxOSOpSf1iOnJzKdtjJo/view?usp=sharing

### CODE REVIEW

File: mason.py
Comment (natolambert) [inline mason.py:1]: Why is this called `mason.py`?

File: rejection_sampling/generation.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+def print_rich_table(df: pd.DataFrame) -> Table:
+    console = Console()
+    table = Table(show_lines=True)
+    for column in df.columns:
+        table.add_column(column)
+    for _, row in df.iterrows():
+        table.add_row(*row.astype(str).tolist())
+    console.print(table)
+
+def main(args: Args, dataset_args: DatasetArgs, gen_args: GenerationArgs):
```
Comment (natolambert) [inline rejection_sampling/generation.py]: I see why you made that minor change in the other PR.

File: rejection_sampling/generation.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    # prompt | completions
+    # -------|------------
+    # q1     | a1
+    # q1     | a2
+    # q1     | a3
+    # q2     | a1
+    # ...
+    table = defaultdict(list)
+    for output, messages in zip(outputs, ds[dataset_args.dataset_train_split]["messages"]):
+        # TODO: filter out duplicate messages ?
```
Comment (natolambert) [inline rejection_sampling/generation.py]: Remove comment or done :)

**Review by natolambert — CHANGES_REQUESTED** [review]
Some high level comments on organization + long term management. 
* Where should we put this directory? 
* Is it good to be separated? Should this live in `open_instruct/*`
* Should add `rejection_sampling/` dir to makefile for style + add license to the top (make it 1st part). If not, we should add a "work in progress" tag to the top of the readme.
* Some comments / TODOs to remove

File: rejection_sampling/generation.py (hunk 2 above)
Comment (vwxyzjn) [reply rejection_sampling/generation.py]: just implemented it

File: mason.py
Comment (vwxyzjn) [reply mason.py:1]: Taken from https://github.com/allenai/mason/tree/main/mason. I had to make some modification to it

**Comment by vwxyzjn** [comment]
Thanks for the review @natolambert, I have moved the script to `open_instruct` and also the bash script to `scripts` folder, which should make more sense.

**Review by natolambert — COMMENTED** [review]
@vwxyzjn were the files `gpt_eval_judge.py` and `gpt_eval_judge1.py` added by mistake?

**Comment by vwxyzjn** [comment]
Yeah that was added by mistake. I added a note on `mason.py`

**Review by natolambert — APPROVED** [review]

**Pull request merged**